components/brave_ads/core/internal/ads/serving/choose/sample_ads.h

```diff
@@ -53,6 +61,20 @@ absl::optional<T> SampleAdFromPredictors(
   return absl::nullopt;
 }
 
+template <typename T>
+std::vector<double> ComputeProbabilities(const std::vector<T>& scores) {
+  std::vector<double> probabilities;
+  probabilities.reserve(scores.size());
+
+  const double normalizing_constant = CalculateNormalizingConstant(scores);
+
+  for (const auto& score : scores) {
```

Comment:
no need in & here. just `for (const auto score : scores) {`.
If we use only `std::vector<int>` type for `scores` then `for (const int score : scores) {` is even better.